pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -459,11 +460,18 @@ func ServerPreferredNamespacedGroupVersionResources(discoveryClient discovery.Di
 	if err != nil {
 		return results, err
 	}
+
+	allErrs := []error{}
 	for _, apiGroup := range serverGroupList.Groups {
-		preferredVersion := apiGroup.PreferredVersion
```

Comment:
#24006 is the server-side fix. But the client needs to handle the old server, thus the compensation here.